Libraries/LibWeb/CSS/StyleValues/CSSLightDark.h

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright (c) 2025, Ladybird contributors
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <LibWeb/CSS/StyleValues/CSSColorValue.h>
+
+namespace Web::CSS {
+
+// https://drafts.csswg.org/css-color-5/#funcdef-light-dark
+class CSSLightDark final : public CSSColorValue {
+public:
+    virtual ~CSSLightDark() override = default;
+
+    static ValueComparingNonnullRefPtr<CSSLightDark> create(RefPtr<CSSStyleValue> light, RefPtr<CSSStyleValue> dark)
+    {
+        return AK::adopt_ref(*new (nothrow) CSSLightDark(move(light), move(dark)));
+    }
+
+    virtual bool equals(CSSStyleValue const&) const override;
+    virtual Color to_color(Optional<Layout::NodeWithStyle const&>) const override;
+    virtual String to_string(SerializationMode) const override;
+
+private:
+    CSSLightDark(RefPtr<CSSStyleValue> light, RefPtr<CSSStyleValue> dark)
+        : CSSColorValue(CSSColorValue::ColorType::LightDark)
+        , m_properties { .light = move(light), .dark = move(dark) }
+    {
+    }
+
+    struct Properties {
+        RefPtr<CSSStyleValue> light;
+        RefPtr<CSSStyleValue> dark;
```

Comment:
These should be nonnull because we know they always have a value. In fact I think for this struct they also want to be `ValueComparingNonnullRefPtr` so that the automatic `operator==` works properly. The constructor and `create()` methods also need to take them as nonnull. I can't remember if the `ValueComparing...` part is necessary there or it'll happen implicitly.

Reply:
Makes sense, done :+1: